recording-oracle/src/modules/liquidity-score/liquidity-score.service.ts

```diff
@@ -83,57 +81,59 @@ export class LiquidityScoreService {
     }
 
     const isCEXCampaign = isCenteralizedExchange(campaign.exchangeName);
+    const windowEnd = new Date();
+    const windowStart = new Date(windowEnd.getTime() - 24 * 60 * 60 * 1000);
 
-    const campaignLiquidityScore: LiquidityScore[] = [];
-
-    const fromDate = campaign.lastSyncedAt;
-    let toDate = campaign.endDate;
-
-    if (new Date() < toDate) {
-      toDate = new Date();
-    }
+    const scores: LiquidityScore[] = [];
+    let allUsersSucceeded = true;
 
     for (const user of campaign.users) {
-      let liquidityScore;
+      const liquidityScore = isCEXCampaign
+        ? await this._calculateCEXLiquidityScore(
+            campaign.exchangeName.toLowerCase(),
+            campaign.token,
+            user,
+            windowStart,
+            windowEnd,
+          )
+        : await this._calculateDEXLiquidityScore(
+            campaign.exchangeName.toLowerCase(),
+            campaign.chainId,
+            campaign.token,
+            user,
+            windowStart,
+            windowEnd,
+          );
 
-      if (isCEXCampaign) {
-        liquidityScore = await this._calculateCEXLiquidityScore(
-          campaign.exchangeName,
-          campaign.token,
-          user,
-          fromDate,
-          toDate,
-        );
-      } else {
-        liquidityScore = await this._calculateDEXLiquidityScore(
-          campaign.exchangeName,
-          campaign.chainId,
-          campaign.token,
-          user,
-          fromDate,
-          toDate,
-        );
+      if (liquidityScore === null) {
+        allUsersSucceeded = false;
+        continue;
       }
 
-      // Round the Liquidity Score
-      liquidityScore = Math.round(liquidityScore);
+      if (!Number.isFinite(liquidityScore) || liquidityScore <= 0) continue;
 
-      if (liquidityScore === 0) {
-        continue;
-      }
+      const rounded = Math.round(liquidityScore);
 
-      await this._saveLiquidityScore(campaign, user, liquidityScore);
+      await this._saveLiquidityScore(
+        campaign,
+        user,
+        rounded,
+        windowStart,
+        windowEnd,
+      );
 
-      campaignLiquidityScore.push({
+      scores.push({
         chainId: campaign.chainId,
         liquidityProvider: user.evmAddress,
-        liquidityScore: liquidityScore.toString(),
+        liquidityScore: String(rounded),
       });
     }
 
-    await this.campaignService.updateLastSyncedAt(campaign, new Date());
+    if (allUsersSucceeded) {
+      await this.campaignService.updateLastSyncedAt(campaign, windowEnd);
+    }
 
-    return campaignLiquidityScore;
+    return scores;
   }
 
   private async _calculateCEXLiquidityScore(
```

Comment:
if the score calculation for a user fails we are still pushing the results and registering it on chain, but lastSyncedAt is not being updated, so it will try to calculate again and upload a new intermediate results again and again while it keeps failing